File: 2020 lipiec/c++/3.cpp
#include <iostream>
#include <fstream>
#include <string>
using namespace std;

string ident[200];
int w[8] = {7, 3, 1, 7, 3, 1, 7, 3};

int main()
{
    ifstream in("identyfikator.txt");
    ofstream out("wyniki4_3.txt");
    
    for (int i = 0; i < 200; i++)
        in >> ident[i];
    
    for (int i = 0; i < 200; i++)
    {
        int suma = 0;
        for (int j = 0; j < 3; j++)
            suma += (ident[i][j] - 'A' + 10) * w[j];
        for (int j = 4; j < ident[i].size(); j++)
            suma += (ident[i][j] - '0') * w[j - 1];
        
        if (suma % 10 != (ident[i][3] - '0'))
            out << ident[i] << '\n';
    }
}

File: 2020 lipiec/python/3.py
with open("identyfikator.txt", "r") as f:
    ident = f.read().strip().split('\n')

w = (7, 3, 1, 0, 7, 3, 1, 7, 3)
alf = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"

with open("wyniki4_3.txt", "w") as f:
    for id in ident:
        suma = 0
        for i, j in enumerate(id):
            if i < 3:
                suma += (alf.find(j) + 10) * w[i]
            else:
                suma += int(j) * w[i]
        
        if suma % 10 != int(id[3]):
            f.write(id + '\n')